JSON/include/Poco/JSON/Array.h

```diff
@@ -26,8 +26,23 @@
 
 
 namespace Poco {
+
 namespace JSON {
 
+class JSON_API Array;
+
+}
+
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+// Explicitly instantiated shared pointer in JSON library
+extern template class Poco::SharedPtr<Poco::JSON::Array>;
+#else
+// Explicitly instantiated shared pointer in JSON library
+extern template class JSON_API Poco::SharedPtr<Poco::JSON::Array>;
```

Comment:
here JSON_API visibilty macro is with non - POCO_OS_FAMILY_WINDOWS. 

But in cpp file its with - POCO_OS_FAMILY_WINDOWS. Is this expected? 

same with Object.h and Object.cpp file.

Reply:
Yes, it is expected. Explicitly instantiating templates is exported differently on Windows.